src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +379,284 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
             {
                 physicalLocation = new PhysicalLocationVersionOne
                 {
-                    Region = CreateRegion(v2Region)
+                    Region = CreateRegion(v2Region, uri: null)
                 };
             }
 
             return physicalLocation;
         }
 
-        internal RegionVersionOne CreateRegion(Region v2Region)
+        internal RegionVersionOne CreateRegion(Region v2Region, Uri uri)
         {
             RegionVersionOne region = null;
 
-            if (v2Region != null && (v2Region.StartColumn > 0 ||
-                                     v2Region.StartLine > 0 ||
-                                     v2Region.EndColumn > 0 ||
-                                     v2Region.EndLine > 0 ||
-                                     v2Region.Length > 0 ||
-                                     v2Region.Offset > 0))
-            {
-                region = new RegionVersionOne
-                {
-                    EndColumn = v2Region.EndColumn,
-                    EndLine = v2Region.EndLine,
-                    Length = v2Region.Length,
-                    Offset = v2Region.Offset,
-                    StartColumn = v2Region.StartColumn,
-                    StartLine = v2Region.StartLine
-                };
+            if (v2Region != null)
+            {
+                region = new RegionVersionOne();
+
+                if (v2Region.StartLine > 0 ||
+                    v2Region.EndLine > 0 ||
+                    v2Region.StartColumn > 0 ||
+                    v2Region.EndColumn > 0 ||
+                    v2Region.CharOffset > 0 ||
+                    v2Region.CharLength > 0)
+                {
+                    if (v2Region.StartLine > 0)
+                    {
+                        // The start of the region is described by line/column
+                        region.StartLine = v2Region.StartLine;
+                        region.StartColumn = v2Region.StartColumn > 0
+                            ? v2Region.StartColumn
+                            : 1;
+                    }
+                    else
+                    {
+                        // The start of the region is described by character offset
+
+                        if (v2Region.CharOffset == 0)
```

Comment:
Nit: This `if` isn't needed because `ConvertCharOffsetToByteOffset` should work for a 0 offset.

Reply:
Done